stream_alert_cli/lookup_tables/handler.py

```diff
@@ -21,33 +24,149 @@
-
     description = 'Describe and manage your LookupTables'
 
     @classmethod
     def setup_subparser(cls, subparser):
-        """Add subparser for LookupTables"""
+        # FIXME (derek.wang) Refactor this into a more robust command-nesting framework
+        subcommands = cls._subcommands()
         set_parser_epilog(
             subparser,
             epilog=(
                 '''\
+Available Sub-Commands:
+
+{}
+
+Examples:
+
+    manage.py lookup-tables [describe-tables|get|set]
+                '''.format('\n'.join([
+                    '\t{command: <{pad}}{description}'.format(
+                        command=command,
+                        pad=30,
+                        description=subcommand.description
+                    )
+                    for command, subcommand
+                    in subcommands.items()
+                ]))
+            )
+        )
+
+        lookup_tables_subparsers = subparser.add_subparsers()
+
+        for _, subcommand in subcommands.items():
+            subcommand.setup_subparser(lookup_tables_subparsers)
+
+    @classmethod
+    def handler(cls, options, config):
+        subcommands = cls._subcommands()
+        if options.subcommand in subcommands:
+            return subcommands[options.subcommand].handler(options, config)
+
+        LOGGER.error('Unhandled lookup-tables subcommand %s', options.subcommand)
+
+    @classmethod
+    def _subcommands(cls):
+        return {
+            # FIXME (derek.wang) Put the command strings into the commands themselves, so the
+            #   subparsers can be registered easily
+            'describe-tables': LookupTablesDescribeTablesSubCommand,
+            'get': LookupTablesGetKeySubCommand,
+            'set': LookupTablesSetSubCommand,
+            'list-add': LookupTablesListAddSubCommand,
+        }
+
+
+class LookupTablesListAddSubCommand(CLICommand):
+    description = 'Add a value to a list-typed LookupTables key'
+
+    @classmethod
+    def setup_subparser(cls, subparser):
+        set_parser = generate_subparser(
+            subparser,
+            'list-add',
+            description='Sets a key on the requested LookupTable',
+            subcommand=True
+        )
+
+        set_parser_epilog(
+            set_parser,
+            epilog=(
+                '''\
                 Examples:
 
-                    manage.py lookup-tables [describe-tables|get|set]
+                    manage.py lookup-tables list-add -t [table] -k [key] -v [value]
                 '''
             )
         )
 
-        lookup_tables_subparsers = subparser.add_subparsers()
+        set_parser.add_argument(
+            '-t',
+            '--table',
+            help='Name of the LookupTable',
+            required=True,
+            default=None
+        )
+
+        set_parser.add_argument(
+            '-k',
+            '--key',
+            help='Key to modify on the LookupTable',
+            required=True,
+            default=None
```

Comment:
reasoning for `default=None` vs the default for this type?

Reply:
_Bad_ copypasta